spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java

```diff
@@ -0,0 +1,106 @@
+
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StoreQueryParameters;
+import org.apache.kafka.streams.state.QueryableStoreType;
+
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.util.Assert;
+
+/**
+ * Provide a wrapper API around the interactive query stores in Kafka Streams.
+ * Using this API, an application can gain access to a named state store in the
+ * {@link KafkaStreams} under consideration.
+ *
+ * @author Soby Chacko
+ * @since 3.2.0
+ */
+public class KafkaStreamsInteractiveQueryService {
+
+	private static final int DEFAULT_MAX_ATTEMPTS = 1;
+	/**
+	 * {@link StreamsBuilderFactoryBean} that provides {@link KafkaStreams} where the state store is retrieved from.
+	 */
+	private final StreamsBuilderFactoryBean streamsBuilderFactoryBean;
+
+	/**
+	 * {@link RetryTemplate} to be used by the interative query service.
+	 */
+	private RetryTemplate retryTemplate = new RetryTemplate();
+
+	/**
+	 * Underlying {@link KafkaStreams} from {@link StreamsBuilderFactoryBean}.
+	 */
+	private KafkaStreams kafkaStreams;
+
+	/**
+	 * Constructs an instance for querying state stores from the KafkaStreams in the {@link StreamsBuilderFactoryBean}.
+	 *
+	 * @param streamsBuilderFactoryBean {@link StreamsBuilderFactoryBean} for {@link KafkaStreams}.
+	 */
+	public KafkaStreamsInteractiveQueryService(StreamsBuilderFactoryBean streamsBuilderFactoryBean) {
+		this.streamsBuilderFactoryBean = streamsBuilderFactoryBean;
+	}
+
+	/**
+	 * Retrieve and return a queryable store by name created in the application.
+	 *
+	 * @param storeName name of the queryable store
+	 * @param storeType type of the queryable store
+	 * @param <T> generic type for the queryable store
+	 * @return queryable store.
+	 */
+	public <T> T retrieveQueryableStore(String storeName, QueryableStoreType<T> storeType) {
+		if (this.kafkaStreams == null) {
+			this.kafkaStreams = this.streamsBuilderFactoryBean.getKafkaStreams();
+		}
+		Assert.notNull(this.kafkaStreams, "KafkaStreams cannot be null");
+		StoreQueryParameters<T> storeQueryParams = StoreQueryParameters.fromNameAndType(storeName, storeType);
+		AtomicReference<StoreQueryParameters<T>> storeQueryParametersReference = new AtomicReference<>(storeQueryParams);
+
+		return getRetryTemplate().execute(context -> {
+			try {
+				return this.kafkaStreams.store(storeQueryParametersReference.get());
+			}
+			catch (Exception e) {
+				throw new IllegalStateException("Error retrieving state store: " + storeName, e);
+			}
+		});
+	}
+
+	private RetryTemplate getRetryTemplate() {
+		if (this.retryTemplate == null) {
+			this.retryTemplate = new RetryTemplate();
+			retryTemplate.setBackOffPolicy(new FixedBackOffPolicy());
```

Comment:
Why do we need these customizations?
Why we cannot rely by default on defaults in `RetryTemplate`?
If it is not what is expected, feel free to inject a custom one.

Therefore we don't need this method at all.
Plus see that you have `this.retryTemplate` initialized in the property definition, so technically we never reach the logic in this method.